### PR TITLE
Feat/broadcast cdn header

### PR DESCRIPTION
**Pull request by NemoOudeis**
# Description
This adds integration with our analytics SDK if it is bundled into the same app.

In addition to that it does:
* extracts the CDN header & content lengtth from network requests and passes it to analytics (not to the perf backend).
* configures proper caching on circleci
* instrumentation of okhttp 2 & 3 calls (okhttp2 calls only partially)
* detect agp version and use `compile` for 2.x and `implementation` for 3.x 

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [ ] I ran `./gradlew check` without errors

### CODE REVIEW

File: performance-tracking-core/src/main/java/com/rakuten/tech/mobile/perf/core/Analytics.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.rakuten.tech.mobile.perf.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Analytics {
+  public static final String CDN_HEADER = "X-CDN-Served-From";
```
Comment (corycaywood) [inline performance-tracking-core/src/main/java/com/rakuten/tech/mobile/perf/core/Analytics.java:8]: This isn't a standard header, right? Should there be a way to specify a different Header key?

File: performance-tracking/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/AnalyticsBroadcaster.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import android.text.TextUtils;
+import com.rakuten.tech.mobile.perf.core.Analytics;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Broadcaster that sends events via local broadcast to peer analytics sdk. */
+class AnalyticsBroadcaster extends Analytics {
+  @VisibleForTesting
+  static final String ACTION = "jp.co.rakuten.sdtd.analytics.ExternalEvent";
```
Comment (corycaywood) [inline performance-tracking/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/AnalyticsBroadcaster.java:18]: It seems really weird to use `jp.co.rakuten.sdtd.analytics.ExternalEvent`.

Should it be something like `com.rakuten.tech.mobile.perf.analytics.ExternalEvent` instead? But that would mean the analytics receiver needs to be updated...

**Review by corycaywood — APPROVED** [review]
LGTM. Just a couple of questions.

File: performance-tracking-core/src/main/java/com/rakuten/tech/mobile/perf/core/Analytics.java (hunk 1 above)
Comment (NemoOudeis) [reply performance-tracking-core/src/main/java/com/rakuten/tech/mobile/perf/core/Analytics.java:8]: currently we don't foresee any change in the header key. if it does we will need to update the SDK and it degrades service gracefully, i.e. continue working just not collecting the header value.

File: performance-tracking/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/AnalyticsBroadcaster.java (hunk 2 above)
Comment (NemoOudeis) [reply performance-tracking/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/AnalyticsBroadcaster.java:18]: it is, but it makes integration easier. 
also a new action would require listening to more actions an the analytics side.

**Pull request merged**